bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleEngineImpl.java

```diff
@@ -220,6 +226,14 @@ protected void receiveCommand(ItemCommandEvent commandEvent) {
         }
     }
 
+    private void receiveThingTrigger(ThingTriggerEvent event) {
+        String channel = event.getSource();
```

Comment:
note my comments above - the event will need a different property to carry the channel information, the source is not the correct one to use.